mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraPerUserMaxQuotaDao.java

```diff
@@ -0,0 +1,125 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.quota;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.james.mailbox.cassandra.table.CassandraDefaultMaxQuota;
+import org.apache.james.mailbox.cassandra.table.CassandraMaxQuota;
+import org.apache.james.mailbox.model.QuotaRoot;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+
+public class CassandraPerUserMaxQuotaDao {
+
+    private final Session session;
+    private final PreparedStatement setMaxStorageStatement;
+    private final PreparedStatement setMaxMessageStatement;
+    private final PreparedStatement getMaxStorageStatement;
+    private final PreparedStatement getMaxMessageStatement;
+    private final PreparedStatement setDefaultMaxStorageStatement;
+    private final PreparedStatement setDefaultMaxMessageStatement;
+    private final PreparedStatement getDefaultMaxStatement;
+
+    @Inject
+    public CassandraPerUserMaxQuotaDao(Session session) {
+        this.session = session;
+        this.setMaxStorageStatement = session.prepare(insertInto(CassandraMaxQuota.TABLE_NAME)
+            .value(CassandraMaxQuota.QUOTA_ROOT, bindMarker())
+            .value(CassandraMaxQuota.STORAGE, bindMarker()));
+        this.setMaxMessageStatement = session.prepare(insertInto(CassandraMaxQuota.TABLE_NAME)
+            .value(CassandraMaxQuota.QUOTA_ROOT, bindMarker())
+            .value(CassandraMaxQuota.MESSAGE_COUNT, bindMarker()));
+        this.getMaxStorageStatement = session.prepare(select(CassandraMaxQuota.STORAGE)
+            .from(CassandraMaxQuota.TABLE_NAME)
+            .where(eq(CassandraMaxQuota.QUOTA_ROOT, bindMarker())));
+        this.getMaxMessageStatement = session.prepare(select(CassandraMaxQuota.MESSAGE_COUNT)
+            .from(CassandraMaxQuota.TABLE_NAME)
+            .where(eq(CassandraMaxQuota.QUOTA_ROOT, bindMarker())));
+        this.getDefaultMaxStatement = session.prepare(select(CassandraDefaultMaxQuota.VALUE)
+            .from(CassandraDefaultMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDefaultMaxQuota.TYPE, bindMarker(CassandraDefaultMaxQuota.TYPE))));
+        this.setDefaultMaxMessageStatement = session.prepare(insertInto(CassandraDefaultMaxQuota.TABLE_NAME)
```

Comment:
It could be great that a DAO is responsible of only one table,
then the link should be done at manager level.

Reply:
true, I will handle both comments in a separate commit, it's a good enhancement